First stab at cleaning up the state singleton

nertpinx opened the pull request:
First some issues are fixed, then `pylint` checks are added so that future commits are checked more thoroughly and last but not least is a conversion for some state attributes to show where fiture cleanup/development is meant to be headed.

This PR builds on top of #68 and that's why the first commit is the same.

ovirt-infra commented:
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

nertpinx commented:
ci test please

ifireball commented:
ci add to whitelist

nyoxi commented on pylintrc:
```diff
@@ -0,0 +1,43 @@
+[MASTER]
+# Use multiple processes to speed up Pylint.  It saves only couple of 6 seconds,
```
"couple o 6"?

nertpinx replied:
s/ 6//, will fix =)

nyoxi commented on pylintrc:
```diff
@@ -0,0 +1,43 @@
+[MASTER]
+# Use multiple processes to speed up Pylint.  It saves only couple of 6 seconds,
+# but the only thing it cannot fetch is duplicate-code, which was mostly
```
"cannot fetch"?
I think I understand what you're trying to express here, but the wording does not make much sense to me.

nertpinx replied:
It's supposed to be "cannot catch", not idea how that one got there.

nyoxi commented on tox.ini:
```diff
@@ -23,11 +24,7 @@ commands = bash -c "\
 [testenv:flake8]
-deps =
-    flake8==3.5
-    pep8-naming==0.8.2
```
where do you get the dependencies from?

nertpinx replied:
the default is little bit more up, by default installing everything from `requirements.txt`. It installs a little bit more than needed, but I think the advantages outweigh that one disadvantage:

 1) the extra stuff is going to be installed for another testenv anyway, so it is cached already,
 1) the dependencies are not listed in two different places,
 1) it makes the `tox.ini` file little bit more concise.

nyoxi commented on .travis.yml:
```diff
@@ -22,7 +22,7 @@ before_install:
 
 install:
-  - pip install tox ansible yamllint
+  - pip install tox ansible
```
I think the reason for installing ansible was to have yamllint (plus something else that I already dropped I think). Is ansible still needed for anything when you add yamllint explicitly? Of course, you may need to change the `ansible --version` to `yamllint --version` below.

nertpinx replied:
I left it there purely for the `ansible --version` to work as I wasn't sure if that was still needed (maybe it is a very important debugging artifact).

nertpinx replied:
I have a cleaner version locally now.

nyoxi commented on tox.ini:
```diff
@@ -15,6 +15,7 @@ deps =
 [testenv:yamllint]
 deps =
     ansible
+    yamllint
```
same here, do we still need ansible?

nertpinx replied:
I tried it without and (maybe because it is being run by bash as the whitelisted external) it does not work. I can have a look at it.

nertpinx replied:
How the hell did that get there? I remember having problems without it, but it does not make sense **at all** to me now. Anyway, thankfully, now I tried with this:
```
[testenv:yamllint]
commands = yamllint -c {toxinidir}/.yamllint {toxinidir}/
```
and it Just Works (TM). So that's going to be in the next version.

nyoxi commented on wrapper/tests/test_state.py:
```diff
@@ -1,39 +1,28 @@
 
 class TestState(unittest.TestCase):
-    """ Tests state object, make sure it it singleton """
+    """ Tests state object, make sure it it state """
```
could you fix the typo (it it) while you're at it, please

nyoxi commented:
It turns out CloudForms is safe so we can do the change. It also turns out that I remembered it slightly wrong and at the end (after `handle_finish`) we do set `fail` to `True` on successful conversion. So all is good.

nertpinx commented:
> I think the approach is great. The only real issue is that it changes the semantics of `failed` little bit. Originally the key existed in state only when the conversion failed (there was never 'failed=false'). If CloudForms checks failure only by existence of the key then we will break things. But for now keep it as you have it.

Even though you later said it works fine, I already have some more features in next patches which only serialize keys that are not `None`. After those it is very easy to just change the initial assignment from ` = False` to ` = None` and that should do the trick.

nertpinx commented:
I rebased this on current master and fixed missing things. Can we push this or would you rather like to see the whole conversion together? If the latter is the case do you want me to add the rest of the commits to this PR?
@nyoxi ^^

nyoxi commented:
Rebased and merged into repository https://github.com/ManageIQ/manageiq-v2v-conversion_host. I am closing the PR.
